### PR TITLE
[manager] Support request Tensors and Weights 

### PR DESCRIPTION
**Pull request by kparichay**
Added support for request Tensors and Weights
Moved spec of Tensors and Weights to VarGrad and Weights correspondingly
Also created constructors for weights and var_grad with their specs

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

File: nntrainer/tensor/manager.cpp [hunk 1]
```diff
@@ -626,4 +626,50 @@ void Manager::deinitializeTensors() {
   tensors_initialized = false;
 }
 
+/**
+ * @brief     Create weights with the given spec
+ *
+ */
+std::vector<Weight *>
+Manager::requestWeights(const GraphNode &node,
+                        std::vector<Weight::Spec> &weights_spec) {
```
Comment (zhoonit) [inline nntrainer/tensor/manager.cpp:635]: how about `const std::vector<Weight::Spec> &` ?

Comment (kparichay) [reply nntrainer/tensor/manager.cpp:635]: this creates issue `reference to local variable` issue.

Comment (zhoonit) [reply nntrainer/tensor/manager.cpp:635]: Right

**Review by zhoonit — APPROVED** [review]
LGTM

File: nntrainer/tensor/manager.cpp (hunk 1 above)
Comment (kparichay) [reply nntrainer/tensor/manager.cpp:635]: i think the copy will be optimized by the compiler.

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Pull request merged**